fluent-bit-to-loki/pkg/client/client.go

```diff
@@ -0,0 +1,76 @@
+package client
+
+import (
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/gardener/logging/fluent-bit-to-loki/pkg/buffer"
+	"github.com/gardener/logging/fluent-bit-to-loki/pkg/config"
+	"github.com/grafana/loki/pkg/promtail/client"
+	"github.com/prometheus/common/model"
+)
+
+type newClientFunc func(cfg client.Config, logger log.Logger) (client.Client, error)
+
+// NewClient creates a new client based on the fluentbit configuration.
+func NewClient(cfg *config.Config, logger log.Logger) (client.Client, error) {
+	var ncf newClientFunc
+
+	if cfg.ReplaceOutOfOrderTS {
+		ncf = newTimestampOrderingClient
+	} else {
+		ncf = client.New
+	}
+
+	if cfg.BufferConfig.Buffer {
+		return buffer.NewBuffer(cfg, logger, ncf)
+	}
+	return ncf(cfg.ClientConfig, logger)
+}
+
+type clientWrapper struct {
+	lokiclient client.Client
+	lastTS     map[model.Fingerprint]time.Time
+	logger     log.Logger
+}
+
+// New makes a new dque loki client
+func newTimestampOrderingClient(cfg client.Config, logger log.Logger) (client.Client, error) {
+	lokiclient, err := client.New(cfg, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return &clientWrapper{
+		lokiclient: lokiclient,
+		lastTS:     make(map[model.Fingerprint]time.Time),
+		logger:     logger,
+	}, nil
+}
+
+// Handle implement EntryHandler; adds a new line to the next batch; send is async.
```

Comment:
I think that not all of this correct. It looks like this function overwrite the timestamp of a log.